src/modules/mavlink/mavlink_main.cpp

```diff
@@ -2744,53 +2746,50 @@ void Mavlink::publish_telemetry_status()
 void Mavlink::configure_sik_radio()
 {
 	/* radio config check */
-	if (_uart_fd >= 0 && _param_sik_radio_id.get() != 0) {
-		/* request to configure radio and radio is present */
-		FILE *fs = fdopen(_uart_fd, "w");
-
-		if (fs) {
-			/* switch to AT command mode */
-			px4_usleep(1200000);
-			fprintf(fs, "+++");
-			fflush(fs);
-			px4_usleep(1200000);
-
-			if (_param_sik_radio_id.get() > 0) {
-				/* set channel */
-				fprintf(fs, "ATS3=%" PRIu32 "\r\n", _param_sik_radio_id.get());
-				px4_usleep(200000);
-
-			} else {
-				/* reset to factory defaults */
-				fprintf(fs, "AT&F\r\n");
-				px4_usleep(200000);
-			}
+	if (_uart_fd < 0 || _param_sik_radio_id.get() == 0) {
+		return;
+	}
 
-			/* write config */
-			fprintf(fs, "AT&W\r\n");
-			px4_usleep(200000);
-
-			/* reboot */
-			fprintf(fs, "ATZ\r\n");
-			px4_usleep(200000);
-
-			// XXX NuttX suffers from a bug where
-			// fclose() also closes the fd, not just
-			// the file stream. Since this is a one-time
-			// config thing, we leave the file struct
-			// allocated.
-#ifndef __PX4_NUTTX
-			fclose(fs);
-#endif
+	// it doesn't seem to switch without this wait
+	px4_usleep(1200000);
```

Comment:
It's necessary to wait 1.2 seconds before doing anything?

If it really does need to be clear for this long I would worry about the potential for the Mavlink RX thread (running at a higher priority) also writing to this.

Reply:
`pthread_mutex_lock(&_send_mutex);`

Reply:
I tried reducing it and it stopped working, so it seems like, yes.

And I did use the lock and then added reading but it all got too complicated, and I gave up on that path.  However, you're right, the mutex would still make sense.